### DISCUSSION TITLE
PascalCase problem with Hasura and Dart

### DISCUSSION THREAD

**Discussion by oriollpz** [dotansimha/graphql-code-generator#10190] [General]
I am using flutter-freezed to transform a Hasura.io GraphQL Schema to a .dart file with Classes and it works good 99%. The problem comes when the classes in dart are created PascalCase, but when they are called inside other classes that they have joins in the database, instead of using PascalCase, they use snake_case.

Example:

The table **users** has a join with the table **user_rooms**. The table **user_rooms** is created as a class with the name UserRooms, but then, the table **users** is created as **Users** and inside this class there is a variable called "userRooms" as a **user_rooms** class, instead of **UserRooms**.

**Comment by oriollpz** [comment]
@Parables

**Comment by Parables** [comment]
Hello @oriollpz , thanks for trying out the flutter-freezed plugin.

Please provide a sample for your schema, your config and the output from the plugin along with your expected output.

Don't forget to tag me @Parables  so that I get notified.

**Comment by oriollpz** [comment]
The Schema is pretty long, I attach it here. It is downloaded from a Hasura.io schema.

There are few problems.

1. The tables with snake_case are created as PascalCase classes but then called snake_case.
2. The models.freezed.dart and models.g.dart are not created.

[Archive.zip](https://github.com/user-attachments/files/17713605/Archive.zip)
